flink-metrics/flink-metrics-datadog/src/test/java/org/apache/flink/metrics/datadog/DatadogHttpReporterFactoryTest.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.metrics.datadog;
+
+import org.apache.flink.metrics.reporter.MetricReporterFactory;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+
+import java.util.ServiceLoader;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import static org.hamcrest.Matchers.hasItem;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Tests for the {@link DatadogHttpReporterFactory}.
+ */
+public class DatadogHttpReporterFactoryTest extends TestLogger {
+
+	@Test
+	public void testMetricReporterSetupViaSPI() {
+		Set<Class<? extends MetricReporterFactory>> loadedFactories = StreamSupport
```

Comment:
What I meant was to have this method:
```
public static void testMetricReporterSetupViaSPI(Class<MetricReporterFactory> clazz) {
	Set<Class<? extends MetricReporterFactory>> loadedFactories = StreamSupport
		.stream(ServiceLoader.load(MetricReporterFactory.class).iterator(), false)
		.map(MetricReporterFactory::getClass)
		.collect(Collectors.toSet());
	assertThat(loadedFactories , hasItems(clazz));
}
```
in flink-metrics-core, and the reporter tests just call it.

Reply:
Hmm, no wonder I feel confused that you did not want the inherent test classes but still feel okay for those redundant code as they are really not much if following this idea.

Sorry for the misunderstanding, and already updated this PR